src/operation-node/query-node.ts

```diff
@@ -43,6 +44,18 @@ export const QueryNode = freeze({
     )
   },
 
+  cloneWithEndModifier<T extends HasEndModifiers>(
+    query: T,
+    modifier: OperationNode,
+  ): T {
+    return freeze({
+      ...query,
+      endModifiers: query.endModifiers
+        ? freeze([...query.endModifiers, modifier])
+        : freeze([modifier]),
+    })
+  },
```

Comment:
nitpicky

let's align with the other methods..

```suggestion
  cloneWithEndModifier<T extends HasEndModifiers>(
    node: T,
    modifier: OperationNode,
  ): T {
    return freeze({
      ...node,
      endModifiers: query.endModifiers
        ? freeze([...node.endModifiers, modifier])
        : freeze([modifier]),
    })
  },
```